lightning/src/blinded_path/message.rs

```diff
@@ -74,16 +91,39 @@ impl Writeable for ForwardTlvs {
 	}
 }
 
-impl Writeable for ReceiveTlvs {
-	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
-		// TODO: write padding
-		encode_tlv_stream!(writer, {
-			(6, self.path_id, option),
-		});
-		Ok(())
-	}
+/// Represents additional data included by the recipient in a [BlindedPath].
+///
+/// This data is encrypted and will be included when sending through
+/// [BlindedPath]. The recipient can authenticate the message and
+/// utilize it for further processing if needed.
+#[derive(Clone, Debug)]
+pub enum RecipientData {
+	/// Represents the data specific to [`OffersMessage`]
+	///
+	/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+	OffersContext(OffersData),
+	/// Represents the data appended with Custom Onion Message.
+	CustomContext(Vec<u8>),
+}
+
+/// Contains the data specific to [`OffersMessage`]
+///
+/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+#[derive(Clone, Debug)]
+pub struct OffersData {
+	/// Payment ID of the outbound BOLT12 payment.
+	pub payment_id: Option<PaymentId>
 }
```

Comment:
We're going to need different variants for storing data in offer paths (see #3117), so let's go ahead and make this an enum and serialize using the corresponding macros. Probably can make an empty variant at least temporarily and another variant with a non-`Option` payment id.

Reply:
Makes sense!
Converted the OffersData into an enum in [653636b](https://github.com/lightningdevkit/rust-lightning/commit/653636b521ec0d4dba73b7aff83e80d594928a9e)
Thanks for the suggestion!